apps/emqx/src/emqx_channel.erl

```diff
@@ -835,6 +801,24 @@ do_subscribe(
             {RC, Channel}
     end.
 
+%% TODO(20231013): rm me
+gen_reason_codes(CheckedTopicFilters, TopicFiltersWithRC) ->
+    TupleTopicFilters2 =
+        lists:foldl(
+            %% previous `deny_subscription` in subopts never used in EMQX 5.x hooks.
```

Comment:
The original idea was to remove the redundant comments after changing the `client.subscribe` hook, since `deny_subscription` and the proposed change in the hook provide the same ability: deny a certain subscription in the hook function.